### DISCUSSION TITLE
Would like a way to disable snake_case to kebab-case conversion for commands and options ...

### DISCUSSION THREAD

**Discussion by breathe** [fastapi/typer#748] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import typer

def main(some_name: str):
   typer.echo(f"Hello {some_name}")

if __name__ == "__main__":
   typer.run(main)
```

### Description

I would like an option to make the cli for the above program be `./entrypoint --some_name foo` rather than `./entrypoint --some-name foo`

#### Use case justification 

As an opinionated full stack developer, I _loathe_ automatic conversions between different 'case' styles for variable names.  This kind of mechanism complicates simplistic code-base search mechanisms unnecessarily.  Consider the reality of modern codebases -- where some command execution is embedded inside some terraform declaration as an arguments array that gets passed into a docker container -- or any other similar varieties of incidental complexity as arises when trying to use modern cloud providers.  While building a company working on a product, I value increasing the likelihood that searching the codebase for `some_name` will find all-at-once the relevant ways in which the parameter is used as well as how the codebase deals with it.  As such I would like the option to globally turn off the case conversions that `typer` does for both commands and options.  I don't particularly care that the cli interface looks like `./entrypoint some_command --some_option` vs `./entrypoint some-command --some-option` as a user of a cli that is internally facing --  but I _do_ want to achieve the quickest visibility into what `some_option` does within my system which includes: where the option gets passed, how the argument is depended on within the implementation, etc ...  

Disabling mechanisms that produce needless aliasing would be useful to me to better achieve that end.

### Operating System

Linux

### Operating System Details

This is not platform dependent

### Typer Version

0.9.0

### Python Version

Python 3.10.10

**Comment by peterjc** [comment]
Duplicate of #341 for minus/hypens vs underscores in command names (which explains some work arounds and that this *is* due to click), and later discussion #1049 focused on option names.
